### ISSUE TITLE
Indentation Check with Namespace

### ISSUE THREAD

**Issue by LupusUmbrae** [wenns/sonar-cxx#441]
I've noticed that the indentation check seems to give incorrect results when indenting after namespaces. I'm expected with indentNamespace = true that the check expects code inside a namespace block to be indented

Take the follow code snippet:

```
  namespace pi {
    using rapidxml::xml_node;
```

With an indentation level of 4, and indentNamespace true i would expect that to pass, instead i get the error "Make this line start at column 1."

Later on when we indent again we get the same issue:

```
    PngDataSet::PngDataSet(...)
    {
        if (config.compare("") != 0)
```

With the above i get the error "Make this line start at column 5."

Thanks in advance

**Comment by jmecosta** [comment]
Yeah I think this check is somehow broken, we had some discussion about
this earlier already . ended up disabling it because cause to much noise

**Comment by LupusUmbrae** [comment]
Had a bit more of a look at this issue and it looks as though in my case the namespace declaration is the first node matched by `translationUnit` (AstScanner line 92), this means that it's the returned `AstNode`.

The top level `AstNode` is only passed in under `visitFile`, which in the indentationCheck doesn't check if it's a declaration that should cause the indentation to change level. Sadly i don't know enough to know if from `visitFile` i can determine what sub-types (i.e `namespaceDeclaration`) the translationUnit has.

Another way to do it would be to do a check on the first node, to check if it's inside a namespace body, i'll try this option and see if i can break it more :)

Hope that informations of some help :)

**Comment by guwirth** [comment]
@Typz code seems to be from you? Do you have time to fix it?

**Comment by LupusUmbrae** [comment]
Did some more investigation, my original thoughts weren't correct. It looks as though our namespace declaration is being parsed to a `recoveredDeclaration tokenValue='namespace' tokenLine=27 tokenColumn=0` which isn't right.

After slowly cutting out bits an pieces of the code it looks like the issue is with `>>`, which i thought there was another issue in but can't find it at the moment

**Comment by markjohndoyle** [comment]
This one?
https://github.com/wenns/sonar-cxx/issues/49

**Comment by guwirth** [comment]
If there is an issue with ```>>``` try the latest 0.9.3. There is a bugfix #378
(please keep in mind that 0.9.2 need SQ 4.5.2 or later).

**Comment by guwirth** [comment]
What should we do with this? Remove the check or fix it?

**Comment by jmecosta** [comment]
I ended up not having any use for it. Duno about others

**Comment by Bertk** [comment]
by the way, why are   namespaceDefinition,   namedNamespaceDefinition not listed in  AstNodeType[] BLOCK_TYPES?

I think we should fix this check.

**Comment by jmecosta** [comment]
@guwirth perhaps this can be moved to another milestone, its low priority

**Comment by Bertk** [comment]
Who can check whether this will fix the issue? 
fix intention check a04c167b7f6b1fb028938a6c7df754020ca4d28c
test data e000bc0f10c45312cadb268c2a56b434104017c3
